pkg/vm/vm.go

```diff
@@ -109,13 +126,24 @@ func NewWithTrigger(t trigger.Type) *VM {
 	vm := &VM{
 		state:   vmstate.None,
 		trigger: t,
+		hooks: defaultHooks,
```

Comment:
I would rather have `!= nil` comparison and remove the whole `defaultHooks` global. It would be nice for default value (from the Go POV) to be valid.

Reply:
see above